src/sinol_make/commands/gen/gen_util.py

```diff
@@ -0,0 +1,138 @@
+import glob
+import os
+
+import argparse
+import signal
+import subprocess
+import sys
+
+from sinol_make import util
+from sinol_make.helpers import compiler, package_util, compile
+
+
+def get_ingen(task_id=None, ingen_path=None):
+    """
+    Find ingen source file in `prog/` directory.
+    If `ingen_path` is specified, then it will be used (if exists).
+    :param task_id: task id, for example abc. If None, then
+                    will return any ingen matching "*ingen.*"
+    :param ingen_path: path to ingen source file
+    :return: path to ingen source file or None if not found
+    """
+
+    if ingen_path is not None:
+        if os.path.exists(ingen_path):
+            return ingen_path
+        else:
+            return None
```

Comment:
I've checked for more places and indeed, there are *a lot* of `None` values in the code. That's not really good error handling. Usually when people prefer to return the errors (instead of throwing an exception), they use a kind of Error object that contains the result or an error message, which can be propagated up. That allows to contain all the logic in the helper functions, instead of cluttering the functions with the business logic and error messages. Though when possible, exiting deep in functions is also OK, as that get rids of the error handling and unnecessary code.

Reply:
I replaced the `None`s with errors in this command. I think that doing it for other commands should be in another PR.